packages/truffle-db/src/loaders/artifacts/test/index.ts

```diff
@@ -0,0 +1,173 @@
+    all: true
+} 
+
+const db = new TruffleDB(config);
+const Migrations = require(path.join(fixturesDirectory, "Migrations.json"));
+const Build = require(path.join(__dirname, "build", "SimpleStorage.json"));
+
+const GetWorkspaceBytecode: boolean = gql`
+query GetWorkspaceBytecode($id: ID!) {
+  workspace {
+    bytecode(id: $id) {
+      id
+      bytes
+    }
+  }
+}`;
+
+describe("Bytecodes", () => {
+  it("loads create bytecodes", async () => {
+    // arrange
+    const expectedId = generateId({ bytes: Migrations.bytecode });
+    const loader = new ArtifactsLoader(db);
+
+    // act
+    await loader.load();
+
+    // assert
+    const {
+      data: {
+        workspace: {
+          bytecode: {
+            bytes
+          }
+        }
+      }
+    } = await db.query(GetWorkspaceBytecode, { id: expectedId });
+
+    expect(bytes).toEqual(Migrations.bytecode);
+  });
+});
+
+
+const GetWorkspaceSource: boolean = gql`
+query GetWorkspaceSource($id: ID!) {
+  workspace {
+    source(id: $id) {
+      id
+      contents
+    }
+  }
+}`;
+
+
+describe("Sources", () => {
+  it("loads contract sources", async () => {
+    // arrange
+    const expectedId = generateId({
+      contents: Migrations.source,
+      sourcePath: Migrations.sourcePath
+    });
+    const loader = new ArtifactsLoader(db);
+
+    // act
+    await loader.load();
+
+    // assert
+    const {
+      data: {
+        workspace: {
+          source: {
+            contents
+          }
+        }
+      }
+    } = await db.query(GetWorkspaceSource, { id: expectedId });
+
+    expect(contents).toEqual(Migrations.source);
+  });
+});
+
+const GetWorkspaceCompilation: boolean = gql`
+query getWorkspaceCompilation($id: ID!) {
+  workspace {
+    compilation(id: $id) {
+      compiler {
+        name
+        version
+      }
+      contracts {
+        name
+        source {
+          contents
+          sourcePath
+        }
+        ast {
+          json
+        }
+      }
+      sources {
+        contents
+        sourcePath
+      }
+    }
+  }
+}`;
+
+describe("Compilation", () => {
+  it("loads compilations", async () => {
+    //arrange
+    const sourceId = generateId({
+      contents: Build.source,
+      sourcePath: Build.sourcePath
+    });
+    const expectedId = generateId({
+      compiler: Build.compiler, 
+      sourceIds: [{ id: sourceId }]
+    });
+    const loader = new ArtifactsLoader(db, compilationConfig);
+
+    //act
+    await loader.load();
+
+    const {
+      data: {
+        workspace: {
+          compilation: {
+            compiler: {
+              version
+            },
+            sources,
+            contracts
+          }
+        }
+      }
+    } = await db.query(GetWorkspaceCompilation, { id: expectedId });
+
+    expect(version).toEqual(Build.compiler.version);
+    expect(sources[0].contents).toEqual(Build.source);
+    expect(contracts[0].name).toEqual(Build.contractName);
+  })
+});
```

Comment:
Can we make this test check to ensure that a single compilation has exactly the correct sources? (no more, no less)

Reply:
Added `expect(sources.length).toEqual(1);` here, let me know if you had something else or more involved in mind!

Reply:
Probably best to:
- make sure this test adds a compilation with more than one source
- check the IDs match

Can we have this test use magic-square, and then check for the three sources in that project?

Reply:
Totally! I had to make some additional changes all around to be sure this works as expected with the mocks, and also I pushed some changes to `solidity-magic-square` so that we'll be able to test for a link reference in the future. The contracts/build files I added here are based on the branch I pushed to `solidity-magic-square`.

Thanks, let me know if this is more what you had in mind!